src/utils/addonCache.ts

```diff
@@ -30,22 +30,33 @@ export async function addToCache(
     await fs.promises.mkdir(cacheFolderPath);
   }
 
-  let cacheFileJSON: { [key: string]: string } = {};
+  let cacheFileJSON: any = {};
   try {
     const cacheFile = await fs.promises.readFile(cacheFilePath, "utf-8");
     cacheFileJSON = JSON.parse(cacheFile);
   } catch (err) {
     console.log("No cache file found");
   }
-  cacheFileJSON[key] = value;
-  await fs.promises.writeFile(cacheFilePath, JSON.stringify(cacheFileJSON));
+
+  let currentLevel = cacheFileJSON;
+  let prevLevel;
+  for (const key of keys) {
+    currentLevel[key] = currentLevel[key] || {};
+    prevLevel = currentLevel;
+    currentLevel = currentLevel[key];
+  }
+
+  const lastKey = keys[keys.length - 1];
+  prevLevel[lastKey] = value;
```

Comment:
This traverses the cache of the guid. It starts by accessing the existing key (the version number) and if there is none, it creates it, then it accesses the next key (file name) and creates it if there is not one for that file name. And so on for however many keys we want. Then we access the last key we added and point it to value, which would be line number pointing to comment.

Reply:
This seems like a common pattern there should already be a solution for (writing a value at a sub key, without having to check if that sub key existed already) but I couldn't find anything in Javascript (python has defaultdict, but even that is only one-level deep).  So it's fine.

Reply:
Does this work?
```suggestion
  let currentLevel = cacheFileJSON;
  
  for (const key of keys) {
    currentLevel = currentLevel[key] = currentLevel[key] || {};
  }
  currentLevel[keys.at(-1)] = value;  
```

note: there's an edge case with both where a parent key exists already, but it contains a string value rather than an object.  e.g. the cache contains `{"3.8.2": {"manifest.json": "a string value"}}`.  You'll have to be careful how it's used.

Reply:
This method seems to be duplicating the first key, which i had an issue with before. 

```
   AssertionError: expected { Object (test-key) } to deeply equal { 'test-key': 'test-value' }
      + expected - actual

       {
      -  "test-key": {
      -    "test-key": "test-value"
      -  }
      +  "test-key": "test-value"
```